### DISCUSSION TITLE
Welcome to shadcn-phone-input Discussions!

### DISCUSSION THREAD

**Discussion by omeralpi** [omeralpi/shadcn-phone-input#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by AdiRishi** [comment]
Happy that you made this. Honestly just wanted to say that this component you made is fantastic! Started using it to build websites, works like a charm 💯

**Comment by bcmendis** [comment]
Hi,
Fantastic component, but Im getting this error when I use the refine function from zod, to validate the input.
Basically the same implementation as OP, but the Zod Schema is abstracted to another file.
Any ideas?

<img width="911" alt="Screenshot 2024-03-09 at 12 54 12 PM" src="https://github.com/omeralpi/shadcn-phone-input/assets/98670180/075015dd-8c15-4318-85a8-46e78b7a5c5d">

**Reply by omeralpi** [reply]
Hi,

If used on the server side, it requires the 'use client' directive. The error can be resolved by adding 'use client;' at the beginning of the file.

Hope this helps!

**Reply by bcmendis** [reply]
Oh my, thank you! It worked.

But I'm wondering why isPossibleNumber and isValidNumber requires the "use client" directive?
They are normal functions, or am I not understanding this library properly? :/

**Reply by omeralpi** [reply]
You're welcome!

See #5.
